server/src/main/java/org/elasticsearch/search/aggregations/metrics/InternalGeoCentroid.java

```diff
@@ -115,7 +115,7 @@ public long count() {
     public InternalGeoCentroid reduce(List<InternalAggregation> aggregations, ReduceContext reduceContext) {
         double lonSum = Double.NaN;
         double latSum = Double.NaN;
-        int totalCount = 0;
+        long totalCount = 0;
```

Comment:
Huh! The `count` member variable was always a long!

Reply:
yup!

Reply:
Huh!